pkg/operation/botanist/component/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh

```diff
@@ -50,9 +41,60 @@ function format-data-device() {
 format-data-device
 {{- end }}
 
-{{ range $name, $image := .images -}}
-docker-preload "{{ $name }}" "{{ $image }}"
-{{ end }}
+LAST_DOWNLOADED_HYPERKUBE_IMAGE=""
+if [[ -f "$PATH_LAST_DOWNLOADED_HYPERKUBE_IMAGE" ]]; then
+  LAST_DOWNLOADED_HYPERKUBE_IMAGE="$(cat "$PATH_LAST_DOWNLOADED_HYPERKUBE_IMAGE")"
+fi
+
+LAST_COPIED_HYPERKUBE_IMAGE=""
+if [[ -f "$PATH_LAST_COPIED_HYPERKUBE_IMAGE" ]]; then
+  LAST_COPIED_HYPERKUBE_IMAGE="$(cat "$PATH_LAST_COPIED_HYPERKUBE_IMAGE")"
+fi
+
+echo "Checking whether we need to preload a new hyperkube image..."
+if [[ "$LAST_DOWNLOADED_HYPERKUBE_IMAGE" != "{{ .hyperkubeImage }}" ]]; then
+  echo "Preloading hyperkube image ({{ .hyperkubeImage }}) because last downloaded image ($LAST_DOWNLOADED_HYPERKUBE_IMAGE) is outdated"
+  {{ .pathDockerBinary }} pull "{{ .hyperkubeImage }}"
+
+  echo "Starting temporary hyperkube container to copy binaries to host"
+
+{{- if semverCompare "< 1.17" .kubernetesVersion }}
+  {{ .pathDockerBinary }} run --rm -v "$PATH_HYPERKUBE_DOWNLOADS":"$PATH_HYPERKUBE_DOWNLOADS":rw "{{ .hyperkubeImage }}" /bin/sh -c "cp /usr/local/bin/kubelet $PATH_HYPERKUBE_DOWNLOADS"
+  {{ .pathDockerBinary }} run --rm -v "$PATH_HYPERKUBE_DOWNLOADS":"$PATH_HYPERKUBE_DOWNLOADS":rw "{{ .hyperkubeImage }}" /bin/sh -c "cp /usr/local/bin/kubectl $PATH_HYPERKUBE_DOWNLOADS"
+{{- else if semverCompare "< 1.19" .kubernetesVersion }}
+  {{ .pathDockerBinary }} run --rm -v "$PATH_HYPERKUBE_DOWNLOADS":"$PATH_HYPERKUBE_DOWNLOADS":rw --entrypoint /bin/sh "{{ .hyperkubeImage }}" -c "cp /usr/local/bin/kubelet $PATH_HYPERKUBE_DOWNLOADS"
+  {{ .pathDockerBinary }} run --rm -v "$PATH_HYPERKUBE_DOWNLOADS":"$PATH_HYPERKUBE_DOWNLOADS":rw --entrypoint /bin/sh "{{ .hyperkubeImage }}" -c "cp /usr/local/bin/kubectl $PATH_HYPERKUBE_DOWNLOADS"
+{{- else }}
+  HYPERKUBE_CONTAINER_ID="$({{ .pathDockerBinary }} run --rm -d -v "$PATH_HYPERKUBE_DOWNLOADS":"$PATH_HYPERKUBE_DOWNLOADS":rw "{{ .hyperkubeImage }}")"
+  {{ .pathDockerBinary }} cp   "$HYPERKUBE_CONTAINER_ID":/kubelet "$PATH_HYPERKUBE_DOWNLOADS"
+  {{ .pathDockerBinary }} cp   "$HYPERKUBE_CONTAINER_ID":/kubectl "$PATH_HYPERKUBE_DOWNLOADS"
+  {{ .pathDockerBinary }} stop "$HYPERKUBE_CONTAINER_ID"
+  chmod +x "$PATH_HYPERKUBE_DOWNLOADS/kubelet"
+  chmod +x "$PATH_HYPERKUBE_DOWNLOADS/kubectl"
+{{- end }}
```

Comment:
Just to be on the safe side, you can always make the copied binaries executable, not only for cluster >=1.19.
```suggestion
{{- end }}
  chmod +x "$PATH_HYPERKUBE_DOWNLOADS/kubelet"
  chmod +x "$PATH_HYPERKUBE_DOWNLOADS/kubectl"
```